nginx/lua/webdav_content.lua

```diff
@@ -0,0 +1,204 @@
+local uv = require('luv')
+local ngx = require('ngx')
+local ffi = require('ffi')
+  ffi.cdef[[
+  int setxattr(const char *path, const char *name, const void *value, size_t size, int flags);
+  ssize_t getxattr(const char *path, const char *name, void *value, size_t size);
+  ]]
+
+--
+-- Probably a naiive implementation of adler32, but it works
+--
+local function adler32_increment(state, buf)
```

Comment:
I was thinking to use the ffi interface to call `libz` adler32 implementation. But this might not be that much slower (or more likely not the limiting factor)

Reply:
yeah, it's worth looking at ... but the implementation was so straightforward I open-coded it here. Honestly, CMS should start recording sha256 or some other type of checksum for the files